fix: toggle linting only when error panel toggled

philippfromme opened the pull request:
* activate and deactivate linting only on toggle error panel
* simplify implementation
* rename #onToggleLinting to #handleToggleLinting
* add linting annotations to Camunda 7 BPMN editor

---

Closes #3127

<!--

Thanks for creating this pull request!

Please make sure to link the issue you are closing as "Closes #issueNr". 
This helps us to understand the context of this PR.

-->

philippfromme commented:
@barmac @pinussilvestrus 🏓

nikku commented on client/src/app/tabs/cloud-bpmn/BpmnEditor.js:
```diff
@@ -483,6 +487,8 @@ export class BpmnEditor extends CachedComponent {
       }
     });
+
+    this.getModeler().get('linting').activate();
```
Duplication with first lines in method.

nikku commented on client/src/app/tabs/cloud-bpmn/BpmnEditor.js:
```diff
@@ -463,6 +463,8 @@ export class BpmnEditor extends CachedComponent {
         }
       });
 
+      this.getModeler().get('linting').activate();
```
I don't understand this duplication.

Why don't we react to a `linting.activated` and trigger the `onLayoutChanged` from there? Overall there seems to be duplication going on here.

philippfromme commented:
@nikku I've simplified the implementation to remove the redundant code. I also noticed that the linting annotations weren't included in the Camunda 7 BPMN modeler yet, so I added them. Once we allow users to add lint rules for both platforms that'll be useful.

nikku reviewed (approved):
Nice follow up improvements!
